.github/workflows/pytest-core-nompi.yml

```diff
@@ -36,102 +35,104 @@ jobs:
            py36-gcc7-openmp,
            py37-gcc7-noopenmp,
            py37-gcc8-openmp,
-           py38-gcc9-openmp
+           py38-gcc9-openmp,
+           py37-osx,
+           docker-devito
         ]
         include:
-            - name: py36-gcc49-openmp
-              python-version: 3.6
-              gcc-version: 4.9
-              openmp: 1
-
-            - name: py37-gcc5-openmp
-              python-version: 3.7
-              gcc-version: 5
-              openmp: 1
-
-            - name: py38-gcc6-openmp
-              python-version: 3.8
-              gcc-version: 6
-              openmp: 1
-
-            - name: py36-gcc7-openmp
-              python-version: 3.6
-              gcc-version: 7
-              openmp: 1
-
-            - name: py37-gcc7-noopenmp
-              python-version: 3.7
-              gcc-version: 7
-              openmp: 0
-
-            - name: py37-gcc8-openmp
-              python-version: 3.7
-              gcc-version: 8
-              openmp: 1
-
-            - name: py38-gcc9-openmp
-              python-version: 3.8
-              gcc-version: 9
-              openmp: 1
+        - name: py36-gcc49-openmp
+          python-version: 3.6
+          os: ubuntu-16.04
+          arch: "gcc-4.9"
+          openmp: 1
+
+        - name: py37-gcc5-openmp
+          python-version: 3.7
+          os: ubuntu-16.04
+          arch: "gcc-5"
+          openmp: 1
+
+        - name: py38-gcc6-openmp
+          python-version: 3.8
+          os: ubuntu-16.04
+          arch: "gcc-6"
+          openmp: 1
+
+        - name: py36-gcc7-openmp
+          python-version: 3.6
+          os: ubuntu-16.04
+          arch: "gcc-7"
+          openmp: 1
+
+        - name: py37-gcc7-noopenmp
+          python-version: 3.7
+          os: ubuntu-16.04
+          arch: "gcc-7"
+          openmp: 0
+
+        - name: py37-gcc8-openmp
+          python-version: 3.7
+          os: ubuntu-16.04
+          arch: "gcc-8"
+          openmp: 1
+
+        - name: py38-gcc9-openmp
+          python-version: 3.8
+          os: ubuntu-16.04
+          arch: "gcc-9"
+          openmp: 1
+
+        - name: py37-osx
+          python-version: 3.7
+          os: macos-latest
+          arch: "osx"
+          openmp: 0
+
+        - name: docker-devito
```

Comment:
Im not sure I like having this build here with the pip install ones. It pollutes the yml with lots of `ifs`. I think it be cleaner as a separate file, thus showing how to setup Docker step by step

Reply:
Then gonna complain there is repeated code, and I think a bash script is not adapted at all to CI.

Reply:
> and I think a bash script is not adapted at all to CI.

You know I agree about this

So yesterday, as I was saying, I was perhaps too tired to realize two things:

- we're talking about how many repeated lines? those for running the tests, and then what?
- there are a few exceptions allowing for *some little* code duplication. Plots, for example, is one of those. I started thinking (this morning, with a fresh mind) that this may well be yet another of those exceptions. 

Thoughts @mloubout @ggorman @navjotk

Reply:
I don't think some duplication is an issue here. This is CI and the more simple and explicit each part of the testing is the easier it is to maintain and debug. Now this way there is less duplication and docker is just another venv  build so erroring/passing is still trackable easily and clearly